### PR TITLE
feat(checkout): CHECKOUT-6277 add eslint config to nx

### PR DESCRIPTION
**Pull request by DeveloperAlan**
## What?
Add `@bigcommerce/eslint-config` to the nx workspace

## Why?
So we can have the same eslint rules applied to all checkout sdk js and tools

## Testing / Proof

Screen Shot 2022-07-11 at 2 16 45 pm

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

File: packages/core/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "extends": "../../tsconfig.base.json",
+    "extends": "../../tsconfig.json",
```
Comment (davidchin) [inline packages/core/tsconfig.json:2]: What's the reason for the rename?

File: package.json [hunk 2]
```diff
@@ -77,10 +78,10 @@
   "devDependencies": {
     "@babel/core": "^7.6.2",
     "@babel/preset-env": "^7.6.2",
-    "@bigcommerce/tslint-config": "^4.0.0",
+    "@bigcommerce/eslint-config": "^1.0.1",
```
Comment (davidchin) [inline package.json:81]: Are you going to do the upgrade in a separate PR?

Comment (DeveloperAlan) [reply package.json:81]: Will do upgrade in a seperate PR since there's about 3000+ errors to fix up

File: packages/core/tsconfig.json (hunk 1 above)
Comment (DeveloperAlan) [reply packages/core/tsconfig.json]: Forgot to remove it from the v2.6.1 lint upgrade, will change it back

File: .eslintrc.json [hunk 3]
```diff
@@ -21,7 +24,10 @@
                     }
                   ]
                 }
-              ]
+              ],
```
Comment (animesh1987) [inline .eslintrc.json:27]: Can we move these configuration to package level? Or for the particular lines. Don't think we want to turn the rule off completely?

File: packages/payment-integration/tsconfig.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 {
-    "extends": "../../tsconfig.base.json",
-  }
+  "extends": "../../tsconfig.base.json",
+}
```
Comment (animesh1987) [inline packages/payment-integration/tsconfig.json:3]: EOL?

File: .eslintrc.json [hunk 5]
```diff
@@ -21,7 +24,10 @@
...
+            "@typescript-eslint/naming-convention": "off",
+            "jsdoc/check-tag-names": "off",
+            "@typescript-eslint/default-param-last": "off"
```
Comment (bc-peng) [inline .eslintrc.json:30]: Rules like `@typescript-eslint/default-param-last` are good for code readability. Could we consider setting it as `warn`?

File: package.json [hunk 6]
```diff
@@ -23,6 +23,7 @@
     "npm": "6"
   },
   "homepage": "https://github.com/bigcommerce/checkout-sdk-js",
+  "prettier": "@bigcommerce/eslint-config/prettier",
```
Comment (bc-peng) [inline package.json:26]: Is this line working? I tried it in `checkout-js`, it was like it did not agree with `"@bigcommerce/eslint-config": "^1.0.1"`.

File: .eslintrc.json (hunk 5 above)
Comment (animesh1987) [reply .eslintrc.json:30]: Yes either that, but depends on how many instances there are. If not many better to turn off on specific lines.

File: package.json (hunk 6 above)
Comment (animesh1987) [reply package.json:26]: Good catch, could be left over from 2.6.1 testing

Comment (DeveloperAlan) [reply package.json]: yep just leftover from 2.6.1

File: .eslintrc.json (hunk 3 above)
Comment (DeveloperAlan) [reply .eslintrc.json]: Sorry these rules were leftovers from the 2.6.1 upgrade, I'll remove these for now

**Review by davidchin — APPROVED** [review]

**Review by animesh1987 — APPROVED** [review]
